calc/calc_mamba_flops.py

```diff
@@ -59,160 +72,196 @@ def config_parser():
-    mamba2_block_flops = 2 * (2 * d_inner + 2 * args.mamba_ngroups  * args.state_size + Nheads) * args.hidden_size * args.batch_size * args.sequence_length # in proj
-    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * (d_inner + 2 * args.mamba_ngroups * args.state_size) * args.conv_dimension * args.d_inner# conv
-    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * d_inner * args.state_size * args.d_inner # dtbx
-    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * d_inner * args.state_size # ssm state rollover
-    mamba2_block_flops += 2 * args.batch_size * args.sequence_length * d_inner * args.state_size * args.d_model # c-> y 
-    mamba2_block_flops += args.batch_size * args.sequence_length * args.hidden_size # z gate output
-    return mamba2_block_flops
+def compute_mamba1_flops(args, iter_factor):
+    d_inner = args.hidden_size * args.expansion_factor
+    dt_rank = math.ceil(args.hidden_size / 16) if args.dt_rank == "auto" else int(args.dt_rank)
+    
+    # SSM state computation
+    ssm_flops = iter_factor * d_inner * (11 * args.state_size + 4 * dt_rank + 1)
+    # Input and output projections
+    mamba_projectors_flops = iter_factor * 6 * d_inner * args.hidden_size 
+    # Convolution operations
+    mamba_conv_flops = iter_factor * 2 * d_inner * args.conv_dimension
+    
+    mamba1_flops = ssm_flops + mamba_projectors_flops + mamba_conv_flops
+    return mamba1_flops * args.tokens
 
-def compute_mamba_flops(args):
+def compute_mamba2_flops(args, iter_factor):
     d_inner = args.hidden_size * args.expansion_factor
-    dt_rank = math.ceil(args.hidden_size / 16) if args.dt_rank == "auto" else args.dt_rank
-    ssm_flops = iter_factor * d_inner * args.tokens * (11 * args.state_size + 4 * dt_rank + 1)
-    mamba_projectors_flops = iter_factor * args.tokens * 6 * d_inner * args.hidden_size 
-    mamba_conv_flops = iter_factor * args.tokens * 2 * d_inner * args.conv_dimension
-    mamba_flops = ssm_flops + mamba_projectors_flops + mamba_conv_flops
-    return mamba_flops
+    Nheads = d_inner // args.mamba_headdim
+    
+    # Input projections
+    mamba2_block_flops = 2 * (2 * d_inner + 2 * args.mamba_ngroups * args.state_size + Nheads) * args.hidden_size
+    # Convolution computations
+    mamba2_block_flops += 2 * (d_inner + 2 * args.mamba_ngroups * args.state_size) * args.conv_dimension + (d_inner + 2 * args.mamba_ngroups * args.state_size)
+    # S4D core computations
+    mamba2_block_flops += 4* d_inner * args.state_size
+    # State updates
+    mamba2_block_flops += 2 * d_inner * args.state_size
+    # Multiply state by C and add Dx
+    mamba2_block_flops += d_inner * (2 + args.state_size)
+    # Gated norm (gate activation + gate-state product + rms norm)
+    mamba2_block_flops += (d_inner + d_inner + 3 * d_inner)
+    # Output projections
+    mamba2_block_flops += 2 * d_inner * args.state_size * args.hidden_size
+    mamba2_block_flops += args.hidden_size
```

Comment:
```suggestion
    mamba2_block_flops += 2 * d_inner * args.hidden_size
```
there is no bias in the out projector